### ISSUE TITLE
Building the project using the stable toolchain

### ISSUE THREAD

**Issue by a-ctor** [OpenByteDev/netcorehost#16]
Hi,

I have just found your project and am trying to use it with the stable toolchain but it won't build. This is because of the features in `lib.rs`:

```
#![feature(maybe_uninit_uninit_array, maybe_uninit_slice, try_trait_v2)]
```

I assume this is intended? It would be nice if it was possible to build the project with the stable toolchain as well. I created a fork where I removed the usages of the features and it seems to work. The changes are about 40 lines of code that are changed removed.

I think the `maybe_uninit_uninit_array` and `maybe_uninit_slice` features can be removed entirely. Maybe it is possible to wrap the `try_trait_v2` in its own feature? This way I could opt out and use the stable toolchain to build.

Thank you for your work by the way works great :)

**Comment by OpenByteDev** [comment]
Its true that the `uninit`features are not needed and are only used for a minor performance uplift. The `try_trait_v2` stuff was mostly done as an experiment.
If you can put the code that uses these features behind a feature flag and make a PR i will merge it. Otherwise I will have a look on my own, when I come around to it.

**Comment by OpenByteDev** [comment]
Published in `0.8.0`
